File: app/src/test/java/com/grace/placessearch/service/local/PlacesTestBase.java
package com.grace.placessearch.service.local;


import com.grace.placessearch.service.PlacesApi;

import java.io.File;

public abstract class PlacesTestBase {

    private static final String RESOURCES_PATH = "/resources";
    private static final PlacesApi PLACES_API = new PlacesApiLocal(getResourcesPath());

    public static final String getResourcesPath() {
        return new File(".").getAbsolutePath() + RESOURCES_PATH;
    }

    protected final PlacesApi getPlaces() {
        return PLACES_API;
    }
}

File: app/src/test/kotlin/com/grace/placessearch/service/local/PlacesTestBase.kt
package com.grace.placessearch.service.local


import java.io.File

abstract class PlacesTestBase {

    protected val places: PlacesApiLocal
        get() = PLACES_API

    companion object {

        private const val RESOURCES_PATH = "/resources"
        private val PLACES_API = PlacesApiLocal(resourcesPath)

        private val resourcesPath: String
            get() = File(".").absolutePath + RESOURCES_PATH
    }
}
